src/tasks.scala

```diff
@@ -261,6 +261,48 @@ object Tasks {
     }
   }
 
+  val duplicateAars = Def.task {
+    Resources
+      .collectdeps(libraryProjects.value)
+      .collect { case a @ AarLibrary(_) ⇒ a }
+      .groupBy(_.moduleID.name)
+      .filter(_._2.length > 1)
+  }
+
+  val checkAarsTaskDef = Def.task {
+    implicit val log = streams.value.log
+    implicit val struct = buildStructure.value
+    val deps = thisProjectRef.value.deepDeps
+    def revDependents(aar: AarLibrary) = deps.filter(_.dependsOnAar(aar))
+    def pkgDependents(data: (String, Seq[AarLibrary])) = {
+      (data._1, data._2, data._2 map(revDependents))
+    }
+    val broken = duplicateAars.value
```

Comment:
Alright, I'm on vacation until Wednesday, will fix it then
On Oct 5, 2015 8:54 PM, "Perry" <notifications@github.com> wrote:

> In src/tasks.scala
> <https://github.com/pfn/android-sdk-plugin/pull/211#discussion_r41182275>:
>
> > +    Resources
> > +      .collectdeps(libraryProjects.value)
> > +      .collect { case a @ AarLibrary(_) ⇒ a }
> > +      .groupBy(_.moduleID.name)
> > +      .filter(_._2.length > 1)
> > +  }
> > +
> > +  val checkAarsTaskDef = Def.task {
> > +    implicit val log = streams.value.log
> > +    implicit val struct = buildStructure.value
> > +    val deps = thisProjectRef.value.deepDeps
> > +    def revDependents(aar: AarLibrary) = deps.filter(_.dependsOnAar(aar))
> > +    def pkgDependents(data: (String, Seq[AarLibrary])) = {
> > +      (data._1, data._2, data._2 map(revDependents))
> > +    }
> > +    val broken = duplicateAars.value
>
> neat, didn't know you could .value on arbitrary task values
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pfn/android-sdk-plugin/pull/211/files#r41182275>.
>